plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/DeploymentConfigListenerJob.java

```diff
@@ -0,0 +1,101 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.core.server.debug;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.jboss.tools.openshift.common.core.connection.IConnection;
+import org.jboss.tools.openshift.common.core.connection.IConnectionsRegistryListener;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IPod;
+
+public class DeploymentConfigListenerJob extends Job implements IConnectionsRegistryListener {
+
+	boolean waitingForDeploymentChange = true;
+	
+	public static final int TIMEOUT = 60_000;//TODO get timeout value from some settings
+
+	private IDeploymentConfig deploymentConfig;
+	
+	private IPod pod = null;
+	
+	private Collection<String> oldPods = Collections.emptySet();
+
+	public DeploymentConfigListenerJob(IDeploymentConfig deploymentConfig) {
+		super("Waiting for OpenShift Pod redeployment");
+		this.deploymentConfig = deploymentConfig;
+		oldPods = ResourceUtils.getPodsForDeploymentConfig(deploymentConfig).stream().map(p -> p.getName()).collect(Collectors.toList());
+	}
+
+	@Override
+	public void connectionAdded(IConnection connection) {
+	}
+
+	@Override
+	public void connectionRemoved(IConnection connection) {
+	}
+
+	@Override
+	public void connectionChanged(IConnection connection, String property, Object oldValue, Object newValue) {
+		if (pod != null) {
+			//we're done already
+			return;
+		}
+		if (waitingForDeploymentChange) {
+			if (deploymentConfig.equals(oldValue) && newValue instanceof IDeploymentConfig) {
+				deploymentConfig = (IDeploymentConfig)newValue;
+				waitingForDeploymentChange = false;
+			}
+			return;
+		}
+		//Wait for new pod once deployment is done
+		if (newValue instanceof IPod) {
+			IPod candidate = (IPod) newValue;
+			String podName = candidate.getName();
+			if (!oldPods.contains(podName) &&
+				"Running".equals(candidate.getStatus()) &&
+				!podName.endsWith("-deploy")) {
```

Comment:
A more exact way of matching the pod would be to check if the selector from the DC would select the pod.  The DC has a spec that is used to create the replicationcontroller which monitors pods to ensure the replica count reaches the desired state.  AFAIK 'deploy' pods do not have the dc selector labels.